lua/luasnip/init.lua

```diff
@@ -711,14 +768,22 @@ local function setup_snip_env()
 
 	setfenv(2, combined_table)
 end
+
+---@alias GetSnipEnv fun(): table
+
+---@type GetSnipEnv
 local function get_snip_env()
 	return session.get_snip_env()
 end
 
+---@type GetIdSnippet
 local function get_id_snippet(id)
 	return snippet_collection.get_id_snippet(id)
 end
 
+---@alias AddSnippets fun(ft: string, snippets: Snippet[], opts: table|nil): nil
```

Comment:
Again, addable vs expandable vs snippet.
I think I'll take care of that distinction after this PR, I pretty much know what changes to make :D

Reply:
Yeah, I read your explanation several times, but I guess I didn't end to understand it well if that didn't click when doing the work.

Reply:
Yeah, it's a pretty fine distinction, and you need quite a bit of context to really motivate why it's useful to have